doc/dev-guide/lvmo-units.md

```diff
@@ -0,0 +1,23 @@
+## LVM Volume Groups
+
+- *lvmVG* reconcile units deploys and manages LVMVolumeGroup CRs
+- The LVMVG resource manager creates individual LVMVolumeGroup CRs for each
+  deviceClass in the LVMCluster CR. The vgmanager watches the LVMVolumeGroup
+  and creates the required volume groups on the individual nodes based on the
+  specified deviceSelector and nodeSelector.
+- The corresponding CRs forms the basis of `vgManager` unit to create volume
+  groups and create lvmd config file
+
+## Openshift SCCs
+
+- When the operator is deployed on an Openshift cluster all the required
+  SCCs are created by `openshiftSccs` reconcile unit
+- The `vg-manager`, `topolvm-node` and `lvmd` containers need elevated
+  permissions to access host LVM commands using `nsenter`.
+
+## Storage Classes
+
+- *topolvmStorageClass* resource units creates and manages all the storage
+  classes corresponding to the deviceClasses in the LVMCluster
+- Storage Class name is generated with a prefix "topolvm-" added to name of the
```

Comment:
Can a storage class name be specified without the topolvm- prefix being added? Admins generally like to control the names of their storage classes.

Reply:
- if admin doesn't want to use a pre-created storage class they are free to create a new storage class and not use pre-created SC at all
- we aren't setting any SC as default but just creating SC's corresponding to device classes

Reply:
What about adding a property on the device class something like `storageClassName` that allows them to override the generated storage class name? It's just a suggestion for a future PR.

Reply:
The intention here is to provide a 1-click install so there is a working storage setup the admin can use from day 1, similar to how OCS does this.  They can create their own sc later if they choose.

Reply:
Upstream users appreciate customizing from the first install. That's the beauty of CRs is that it allows for the customization from the start. Admins don't generally want to have extra/unused/uncustomizable storage classes. 1-click install is great for the common case, but requiring the 1-click install doesn't work well for the advanced users.

Reply:
A good point and can be taken up in later improvements. Out first priority was to have a working operator and one of the requirements was a one-click installation.